autometa/validation/datasets.py

```diff
@@ -0,0 +1,174 @@
+GNU Affero General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License
+along with Autometa. If not, see <http://www.gnu.org/licenses/>.
+COPYRIGHT
+
+pulling data from google drive folder with simulated or synthetic communities
+"""
+
+
+import gdown
+import os
+import logging
+import pandas as pd
+
+logger = logging.getLogger(__name__)
+
+# add function to get the file ids from the user input
+def fetchIDs(folder: str, file: str):
+    """
+    Parameters
+    ----------
+    folder : str
+        specifies the folder that the user would like to download from, if any
+    file : str
+        specifies the file that the user would like to download, if any
+
+    Returns
+    -------
+    dict
+        nested dictionary with the user's specified files' folder, file, and file_id
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+
+    """
+    # create the dataframe here
+    index = pd.read_csv("gdown_fileIDs.csv", dtype=str)
+
+    # retrieve only the entries that the user wants
+    if folder == None:
+        target_folder = index
+    else:
+        target_folder = index.query(f'folder == "{folder}"')
+
+    if file == None:
+        target_files = target_folder
+    else:
+        target_files = target_folder.query(f'file == "{file}"')
+
+    targets_dict = target_files.to_dict()
+    return targets_dict
+
+
+def download(targets_dict, out_dirpath: str) -> None:
+
+    """Downloads the files specified in a dictionary.
+
+    Parameters
+    ----------
+    targets_dict : dict
+        nested dictionary with the user's specified files' folder, file, and file_id
+        format: {'folder': {<index>: <folder>}, 'file': {<index>: <file>}, 'file_id': {<index>: <file_id>}}
+    out_dirpath : str
+        directory path where the user wants to download the file(s)
+
+    Returns
+    -------
+    None
+        download is completed through gdown
+
+    """
+
+    key_list = [*targets_dict["file"]]
+    for key in key_list:
+        # retrieve values from targets_dict
+        file = targets_dict["file"][key]
+        folder = targets_dict["folder"][key]
+        file_id = targets_dict["file_id"][key]
+
+        # construct file id into a url to put into gdown
+        url = f"https://drive.google.com/uc?id={file_id}"
+        filename = f"{folder}_{file}"
+        out_fpath = os.path.join(
+            out_dirpath, filename
+        )  # this returns an error when relative file paths get used, eg "~/download"
+
+        # download the specified file with gdown
+        gdown.download(url, out_fpath)
+
+
+def main():
+    import argparse
+    import logging as logger
+
+    logger.basicConfig(
+        format="[%(asctime)s %(levelname)s] %(name)s: %(message)s",
+        datefmt="%m/%d/%Y %I:%M:%S %p",
+        level=logger.DEBUG,
+    )
+
+    parser = argparse.ArgumentParser(
+        prog="autometa-download-dataset",
```

Comment:
The entrypoint should be specified in `setup.py`

```suggestion
```

Reply:
This should be good now? I'd appreciate you giving it a look to make sure I didn't break anything. First time under the hood and all...